pyproject.toml

```diff
@@ -29,10 +29,11 @@ keywords = [
 
 dependencies = [
     'numpy>=1.20.0',
-    'pandas<2.1',
+    'pandas',
     'mdanalysis>=2.0.0',
     'pytest',
     'matplotlib',
+    'setuptools',
```

Comment:
Huh setuptools as a runtime dependency is really rare - what are you using it for? Usually it goes under build time dependencies.

P.S. if you drop versioneer you could drop setup.py completely I think

Reply:
using `pkg_resources` in python 3.12 has an explicit dependency on setuptools now. IIRC there is a replacement but I was too lazy to search it out.

Reply:
I always mess up the `versioningit` setup for some reason.

Reply:
Ah yeah - I'd drop `pkg_resources`, best thing is to use importlib. Here's the giant MDA example 😅 : https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/MDAnalysisTests/datafiles.py

Reply:
re: versioningit setup - here's the recent change we did for waterdynamics, hopefully should give you all the things that need changing: https://github.com/MDAnalysis/waterdynamics/pull/28/files